lib/vtls/openssl.c

```diff
@@ -4256,6 +4276,48 @@ static CURLcode servercert(struct Curl_cfilter *cf,
       infof(data, " SSL certificate verify ok.");
   }
 
+  verify_result = SSL_get_verify_result(backend->handle);
+  if(verify_result != X509_V_OK)
+    certstack = SSL_get_peer_cert_chain(backend->handle);
+  else
+    certstack = SSL_get0_verified_chain(backend->handle);
+  num_cert_levels = sk_X509_num(certstack);
+  OpenSSL_add_all_algorithms();
+  OpenSSL_add_all_digests();
+
+  for(cert_level = 0; cert_level < num_cert_levels; cert_level++) {
+    char cert_algorithm[80];
+    char group_name[80];
+    char group_name_final[80];
+    const X509_ALGOR *palg_cert = NULL;
+    const ASN1_OBJECT *paobj_cert = NULL;
+    X509 *current_cert = NULL;
+    EVP_PKEY *current_pkey;
+    int key_bits;
+    int key_sec_bits;
+    int get_group_name;
+
+    current_cert = sk_X509_value(certstack, cert_level);
+
+    X509_get0_signature(NULL, &palg_cert, current_cert);
+    X509_ALGOR_get0(&paobj_cert, NULL, NULL, palg_cert);
+    OBJ_obj2txt(cert_algorithm, sizeof(cert_algorithm), paobj_cert, 0);
+
+    current_pkey = X509_get0_pubkey(current_cert);
+    key_bits = EVP_PKEY_bits(current_pkey);
+    key_sec_bits = EVP_PKEY_get_security_bits(current_pkey);
+    get_group_name = EVP_PKEY_get_group_name(current_pkey, group_name,
```

Comment:
I think the return code `get_group_name` needs to be checked.
It may be null.

This results in the valgrind errors reported in the build ci ie:
```
 ==112720== Conditional jump or move depends on uninitialised value(s)
 ==112720==    at 0x48D1EBA: dprintf_formatf (mprintf.c:839)
 ==112720==    by 0x48D28F8: curl_mvsnprintf (mprintf.c:1036)
 ==112720==    by 0x48D2A30: curl_msnprintf (mprintf.c:1056)
 ```
 
 If the RC is checked and the get fails you can opt to put null in the message or similar...

Reply:
I think you need to make sure to clear it in declaration so that if OpenSSL does not touch it, it is blank and not containing random stack content. I mean like:

~~~c
 char group_name_final[80] = "";

Reply:
addressed in recent commit per @bagder suggestion I initialized all vars that are read without an explicit initialization